lib/helpers/model/castBulkWrite.js

```diff
@@ -190,7 +190,7 @@ module.exports = function castBulkWrite(originalModel, op, options) {
 
       // set `skipId`, otherwise we get "_id field cannot be changed"
       const doc = new model(op['replaceOne']['replacement'], strict, true);
-      if (model.schema.options.timestamps) {
+      if (model.schema.options.timestamps && getTimestampsOpt(op['replaceOne'], options)) {
```

Comment:
I'd really prefer to inline this so we don't have an extra function here, could just do `op['replaceOne'].timestamps ?? options.timestamps ?? true`